core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCatalog.java

```diff
@@ -525,6 +525,34 @@ public void testVersionHintFileMissingMetadata() throws Exception {
         () -> TABLES.load(tableLocation));
   }
 
+  @Test
+  public void testTableEquality() throws Exception {
```

Comment:
Would it make sense to test that two tables with different properties but the same table identifier are not equal?

Reply:
The main motivation to implement `equals` and `hashCode` was to refresh the cache in Spark. Therefore, I did not consider anything except the identifier so that if a table property was modified, we would still invalidate the cache as both entries point to the same table.

After thinking about this more, modifying `equals` and `hashCode` in `BaseTable` may not be the best idea. Instead, we can simply compare `icebergTable.toString()` in `SparkTable` and limit this equality on table identifiers only to Spark.

Reply:
What do you think about this idea, @holdenk @RussellSpitzer @rdblue?

Reply:
We did a similar thing in #1512.

Reply:
Sounds like a good idea. I like to avoid implementing the method in places where "equals" has a fuzzy definition -- is it the "same" table or is it an equivalent instance?

We're also relying on `Table.toString` behavior a bit too much. It is probably time to add `Table.name` to the API and use that instead.

Reply:
Sounds reasonable to me

Reply:
sgtm

Reply:
It may be a good point in time to implement #658. Let me invest into that.

Reply:
I've created #1537.

Reply:
Throwing in my +1 for adding `name` to the API. But I'd hope that we can still leave a meaningful enough `toString` method in place.

Without it, we just wind up with your typical memory location etc, specifically when / if `Table` is used in a parameterized test (I just spent half my evening adding parameterized names to all of the tests that didn't have them).

What is everybody's opinion on implementing `toString` when it's not strictly necessary? Would you say it's a good practice, or is it possibly just extra unused code that needs to be maintained over time? I know this is potentially more of a religious question than a hard and fast rule, but I'd love to hear your input.

Reply:
I think implementing `toString` is a good practice. Spark uses it in query plans quite a bit.